app/lib/foreman_webhooks/renderer/preview_mock.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module ForemanWebhooks
+  module Renderer
+    class PreviewMock < BasicObject
+      def initialize(name = '@object')
+        @name = name
+      end
+
+      def method_missing(name, *_args, &_block)
+        PreviewMock.new("#{@name}.#{name}")
+      end
+
+      def respond_to_missing?(_method_name, _include_private = false)
+        true
+      end
+
+      def to_s
+        @name
+      end
+
+      def to_jail
+        self
+      end
+
+      def to_json(*_args)
+        to_s
+      end
+
+      def as_json(*_args)
+        to_s
+      end
+
+      def instance_eval
+        to_s
+      end
+
+      def instance_exec
+        to_s
+      end
```

Comment:
Unfortunately, Ruby has a hardcoded warning that cannot be removed: foreman_webhooks/renderer/preview_mock.rb:42: warning: redefining `__send__' may cause serious problems

I tried to use `alias` or `alias_method` without any luck, the warning is still there.

I do not understand why we need to override? It works as expected without redefining `__send__`:

```
[1] pry(main)> object = ForemanWebhooks::Renderer::PreviewMock.new
=> @object.inspect@object.inspect
[2] pry(main)> object.__send__(:instance_exec) { 1 + 1 }
=> "@object"
```

Reply:
I think we should not override `__send__` the pure existance of this method is to avoid being overriden. If I understand Marek correctly, this was an idea just to prevent when a new method is added to the BasicObject class. I think we can live with that.

I suggest to remove the override and add a unit test that will make sure that Ruby BasicObject contains just the methods we expect.

```
[12] pry(main)> BasicObject.instance_methods
=> [:__send__, :!, :==, :!=, :equal?, :__id__, :__binding__, :instance_eval, :instance_exec]
```